modules/apps/data-engine/data-engine-rest-test/src/testIntegration/java/com/liferay/data/engine/rest/resource/v2_0/test/DataRecordResourceTest.java

```diff
@@ -469,6 +471,52 @@ public void testPostDataRecordCollectionDataRecord() throws Exception {
 			404,
 			dataRecordResource.postDataRecordCollectionDataRecordHttpResponse(
 				RandomTestUtil.randomLong(), randomDataRecord()));
+
+		// --> Persist data for text in a fieldset
+
+		DataDefinition postedDataDefinition =
+			DataDefinitionTestUtil.addDataDefinitionWithFieldSet(
+				testGroup.getGroupId());
+
+		DataRecord dataRecord = new DataRecord() {
+			{
+				dataRecordValues = HashMapBuilder.<String, Object>put(
+					"Text",
+					HashMapBuilder.<String, Object>put(
+						"en_US", "Text Value"
+					).build()
+				).put(
+					"fieldset1",
+					HashMapBuilder.<String, Object>put(
+						RandomTestUtil.randomString(),
+						HashMapBuilder.<String, Object>put(
+							"Text",
+							HashMapBuilder.<String, Object>put(
+								"en_US", "Text Value"
+							).build()
+						).build()
+					).build()
+				).build();
+			}
+		};
+
+		List<DDLRecordSet> dataDefinitionRecordSets =
```

Comment:
rename to ddlRecordSetList

Reply:
I'm in doubt about this on. What do you think @gabrielwas ?

Reply:
Actually, I think we could call `postDataDefinitionDataRecord` (passing `dataDefinitionId`) instead of calling `postDataRecordCollectionDataRecord `. That away, we can suppress these lines.

Reply:
I tend to disagree on this one. The method I'm testing is `postDataRecordCollectionDataRecord` according to the name of the test method. The way I see it, if I'm going to test data record persistence calling `postDataDefinitionDataRecord`, I should be writing this code elsewhere.

Does it make sense?

Reply:
You're right. Or maybe we could place this test case inside `testPostDataDefinitionDataRecord`. What do you think?

Reply:
I believe this scenario was already covered for this endpoint. I can double-check it anyway!

Reply:
Okay, checked it out here. No fieldset data record test. And testing `postDataDefinitionDataRecord` would automatically test `postDataRecordCollectionDataRecord` since one calls another.

I'll transfer and adapt the test for `testPostDataDefinitionDataRecord`

Reply:
Great! Thanks, Alejo!